src/officehours_api/views.py

```diff
@@ -53,17 +53,34 @@ class UserList(DecoupledContextMixin, generics.ListAPIView):
     serializer_class = ShallowUserSerializer
 
 
-class UserDetail(DecoupledContextMixin, generics.RetrieveAPIView):
+class UserDetail(DecoupledContextMixin, LoggingMixin, generics.RetrieveUpdateAPIView):
     queryset = User.objects.all()
     permission_classes = (IsAuthenticated,)
 
-    def get(self, request, *args, **kwargs):
+    def get_serializer(self, instance, data=None, many=None, partial=None):
+        ctx = self.get_serializer_context()
+        kwargs = {}
+        if data:
+            kwargs['data'] = data
+        return (
+            MyUserSerializer(instance, context=ctx, **kwargs)
+            if instance == ctx['user']
+            else ShallowUserSerializer(instance, context=ctx, **kwargs)
+        )
+
+    def check_change_permission(self, request, user):
+        if user != request.user:
+            self.permission_denied(request)
+
+    def update(self, request, *args, **kwargs):
         user = self.get_object()
-        if user == request.user:
-            serializer = MyUserSerializer(user, context={'user': request.user})
-        else:
-            serializer = ShallowUserSerializer(user, context={'user': request.user})
-        return Response(serializer.data)
+        self.check_change_permission(request, user)
+        return super().update(request, *args, **kwargs)
+    
+    def partial_update(self, request, *args, **kwargs):
+        user = self.get_object()
+        self.check_change_permission(request, user)
+        return super().update(request, *args, **kwargs)
```

Comment:
These are basically identical. I'm assuming we need both?

Reply:
Thanks for pointing that out, I noticed that in the latter one, it should be calling `super().partial_update`.